tests/integration_tests/tests/agentless_tests/test_blueprint_upload.py

```diff
@@ -284,3 +287,65 @@ def _assert_cleanup(self, blueprint_id):
         )
         self.assertEqual(resp.status_code,
                          requests.status_codes.codes.not_found)
+
+
+class BlueprintImportedTest(AgentlessTestCase):
+
+    def test_blueprints_separate_delete_success(self):
+        first_blueprint_id = 'first'
+        first_blueprint_path = os.path.join('dsl', 'simple_deployment.yaml')
+        self.client.blueprints.upload(resource(first_blueprint_path),
+                                      entity_id=first_blueprint_id)
+        wait_for_blueprint_upload(first_blueprint_id, self.client)
+        second_blueprint_id = 'second'
+        second_blueprint_path = os.path.join('dsl', 'simple_deployment.yaml')
+        self.client.blueprints.upload(resource(second_blueprint_path),
+                                      entity_id=second_blueprint_id,
+                                      async_upload=True)
+        self.client.blueprints.delete(first_blueprint_id)
+        wait_for_blueprint_upload(second_blueprint_id, self.client)
+        self.client.blueprints.delete(second_blueprint_id)
+
+    def test_blueprints_imported_upload_success(self):
+        imported_blueprint_id = 'imported'
+        imported_blueprint_path = os.path.join('dsl', 'simple_deployment.yaml')
+        self.client.blueprints.upload(resource(imported_blueprint_path),
+                                      entity_id=imported_blueprint_id)
+        wait_for_blueprint_upload(imported_blueprint_id, self.client)
+        main_blueprint_id = 'main'
+        main_blueprint_path = \
+            os.path.join('dsl', 'simple_deployment_with_import.yaml')
+        self.client.blueprints.upload(resource(main_blueprint_path),
+                                      entity_id=main_blueprint_id,
+                                      async_upload=True)
+        wait_for_blueprint_upload(main_blueprint_id, self.client)
```

Comment:
surely this is equivalent to just not having async_upload, and then you dont need the wait()?

Reply:
Yeah, that's updated too (looked at it with fresh eyes this morning)